src/libstd/sync/mod.rs

```diff
@@ -10,10 +10,125 @@
 
 //! Useful synchronization primitives.
 //!
-//! This module contains useful safe and unsafe synchronization primitives.
-//! Most of the primitives in this module do not provide any sort of locking
-//! and/or blocking at all, but rather provide the necessary tools to build
-//! other types of concurrent primitives.
+//! ## The need for synchronization
+//!
+//! On an ideal single-core CPU, the timeline of events happening in a program
```

Comment:
You can have concurrency on a single core CPU.

Reply:
I do mention this possibility when talking about the need for compiler fences with signal handlers / interrupts in single-threaded programs.
I couldn't think of a better way to introduce the idea of sequential consistency / program order, in comparison to actual execution order (which is why I used "ideal", to indicate reality is a bit more complicated).

Reply:
So, I get where this paragraph is coming from, but agree it could be worded differently. I'm struggling a bit... I think the issues are:

1. it's not clear that this is "ideal".
2. This is phrased as being true, when the intention is to show that it's not true.